Library/Homebrew/hardware.rb

```diff
@@ -186,6 +203,7 @@ def in_rosetta2?
   end
 
   class << self
+    sig { returns(T.nilable(T.any(Integer, String))) }
```

Comment:
```suggestion
    sig { returns(String) }
```

If you also change the `else` clause to be `Hardware::CPU.cores.to_s` which would makes sense to me.
(can't include that in the suggestion as it's outside the diff range)